pkg/podres/middleware/terminalpods/terminalpods.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package terminalpods
+
+import (
+	"context"
+	"k8s.io/klog/v2"
+
+	"google.golang.org/grpc"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1"
+)
+
+const terminalFieldSelector = "status.phase==Failed,status.phase==Succeeded"
+
+type filteringClient struct {
+	debug bool
+	cli   podresourcesapi.PodResourcesListerClient
+	kcli  *kubernetes.Clientset
+}
+
+func (fc *filteringClient) List(ctx context.Context, in *podresourcesapi.ListPodResourcesRequest, opts ...grpc.CallOption) (*podresourcesapi.ListPodResourcesResponse, error) {
+	resp, err := fc.cli.List(ctx, in, opts...)
+	if err != nil {
+		return resp, err
+	}
+	pods, err := terminalPods(ctx, fc.kcli)
+	if err != nil {
+		return resp, err
+	}
+	FilterFrom(resp, pods)
+	return resp, nil
+}
+
+func (fc *filteringClient) GetAllocatableResources(ctx context.Context, in *podresourcesapi.AllocatableResourcesRequest, opts ...grpc.CallOption) (*podresourcesapi.AllocatableResourcesResponse, error) {
+	resp, err := fc.cli.GetAllocatableResources(ctx, in, opts...)
```

Comment:
just
```
return fc.cli.GetAllocatableResources(ctx, in, opts...)
```
?

Reply:
The IDE complained for some reason, i'll check again, maybe it's a glitch